src/hooks/useWalletTokenAsset.js

```diff
@@ -0,0 +1,20 @@
+import { find, matchesProperty } from 'lodash';
+import { useMemo } from 'react';
+import useAccountAssets from './useAccountAssets';
+import { AssetTypes } from '@rainbow-me/entities';
+
+// this is meant to be used for assets under balances
+// NFTs are not included in this hook
+export default function useWalletTokenAsset(type, uniqueId) {
+  const { allAssets } = useAccountAssets();
+
+  return useMemo(() => {
+    if (!uniqueId) return null;
+
+    let matched = null;
+    if (type === AssetTypes.token) {
+      matched = find(allAssets, matchesProperty('uniqueId', uniqueId));
```

Comment:
Since it can run quite often right now we can use the fastest way for this loop - good old for loop;

```suggestion
      for (let i = 0; i < allAssets.length; i++) {
        const item = allAssets
        
        if (item.uniqueId === uniqueId) {
          matched = item;
          break;
        }
      }
```

Reply:
hey this PR is in progress - this will be replaced with just an index into the accountAssetsData map! no loop required